client/src/app/modules/project-page/components/project-page.component.spec.ts

```diff
@@ -1,56 +1,81 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { LoadingIndicatorModule } from '../../loading-indicator/loading-indicator.module';
-import { MatIconModule } from '@angular/material/icon';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatInputModule } from '@angular/material/input';
-import { API_PROJECT_URL, API_PROJECT_DETAIL_URL } from '../../../tokens';
 import { MatTooltipModule } from '@angular/material/tooltip';
-import { CommonModule } from '@angular/common';
 import { MatButtonModule } from '@angular/material/button';
 import { MatTableModule } from '@angular/material/table';
 import { MatSortModule } from '@angular/material/sort';
 import { ClipboardModule } from '@angular/cdk/clipboard';
 import { NotificationModule } from '../../notification/notification.module';
+import {
+  API_ALL_QUICKSTARTERS_URL,
+  API_GENERATE_PROJECT_KEY_URL,
+  API_PROJECT_DETAIL_URL,
+  API_PROJECT_TEMPLATES_URL,
+  API_PROJECT_URL
+} from '../../../tokens';
 import { BrowserService } from '../../browser/services/browser.service';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { StorageService } from '../../storage/services/storage.service';
+import { MatIconTestingModule } from '@angular/material/icon/testing';
 
-/* TODO fixme + add tests */
 describe('ProjectPageComponent', () => {
-  const createComponent = createComponentFactory({
-    component: ProjectPageComponent,
-    imports: [
-      CommonModule,
-      RouterTestingModule,
-      HttpClientTestingModule,
-      LoadingIndicatorModule,
-      MatIconModule,
-      MatFormFieldModule,
-      MatInputModule,
-      MatTooltipModule,
-      MatButtonModule,
-      MatTableModule,
-      MatSortModule,
-      ClipboardModule,
-      NotificationModule
-    ],
-    providers: [
-      { provide: API_PROJECT_DETAIL_URL, useValue: '/api/mock' },
-      { provide: API_PROJECT_URL, useValue: '/api/mock' },
-      {
-        provide: BrowserService,
-        useValue: {
-          scrollIntoViewById: jest.fn()
+  let component: ProjectPageComponent;
+  let fixture: ComponentFixture<ProjectPageComponent>;
+
+  let mockBrowserService;
+  let mockStorageService;
+
+  beforeEach(async () => {
+    mockStorageService = jasmine.createSpyObj('StorageService', ['getItem']);
+
+    await TestBed.configureTestingModule({
+      declarations: [ProjectPageComponent],
+      imports: [
+        ReactiveFormsModule,
+        FormsModule,
+        RouterTestingModule,
+        HttpClientTestingModule,
+        LoadingIndicatorModule,
+        MatIconTestingModule,
+        MatFormFieldModule,
+        MatInputModule,
+        MatTooltipModule,
+        MatButtonModule,
+        MatTableModule,
+        MatSortModule,
+        ClipboardModule,
+        NotificationModule
+      ],
+      providers: [
+        { provide: API_PROJECT_DETAIL_URL, useValue: '/api/mock' },
```

Comment:
I saw this block with `API_...` in a couple of spec files. Maybe it makes sense to put it somewhere more central. Would it work to have something like a common test module e.g. with all the `Mat...` modules and this block in the `providers` section? With this it would be sufficient to only add this new testing module in every spec file.